### PR TITLE
Refactor: replace (some) dotimes with dolist/seq-map-indexed.

### PR DESCRIPTION
**Pull request by PhilHudson**

### CODE REVIEW

File: blackjack.el [hunk 1]
```diff
@@ -676,46 +674,45 @@ Can be a single-character currency symbol such as \"$\", \"€\" or \"£\", or a
...
-        (setq suit (slot-value card 'suit)))
-      (insert (blackjack--card-face game value suit))
-      (insert " "))
+    (seq-map-indexed
+     (lambda (card index)
+       "Display one of the cards in the dealer's hand."
+       (insert
+        (apply #'blackjack--card-face game
+               (if (and (= index 0) (slot-value dealer-hand 'hide-down-card))
+                   `(,blackjack--card-value-hidden 0)
```
Comment (gdonald) [inline blackjack.el:690]: Right, because the suit does not matter on the dealer's down card.

File: blackjack.el [hunk 2]
```diff
@@ -676,46 +674,45 @@ Can be a single-character currency symbol such as \"$\", \"€\" or \"£\", or a
...
 
 (defun blackjack--draw-player-hands (game)
   "Draw GAME players hands."
-  (let ((player-hands (slot-value game 'player-hands))
-        (player-hand))
-    (dotimes (x (length player-hands))
-      (setq player-hand (nth x player-hands))
-      (blackjack--draw-player-hand game player-hand x))))
+  (seq-map-indexed
+   (apply-partially #'blackjack--draw-player-hand game)
```
Comment (gdonald) [inline blackjack.el:714]: How can this call to `blackjack--draw-player-hand` continue to work properly without the index value being passed?

Comment (PhilHudson) [reply blackjack.el:714]: The index value is in fact being passed, "behind the scenes". Unlike other `*map*` functions, `seq-map-indexed` needs a function that takes *two* arguments, the second of which is the index. You don't have to declare or initialize the index, that's all taken care of for you by `seq-map-indexed`. `(info "(elisp) Sequence Functions")`.

File: blackjack.el (hunk 1 above)
Comment (PhilHudson) [reply blackjack.el:690]: I may have got that right by accident then. I think I may have just forgotten to take care of setting the suit to zero.

File: blackjack.el (hunk 2 above)
Comment (gdonald) [reply blackjack.el:714]: Ok, I see.  Thanks for explaining.

**Comment by gdonald** [comment]
Thank you.

**Pull request merged**